app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -16,24 +16,27 @@ def find_destination_in_vmdb(vm_uid_hash)
 
   def gather_storage_account_properties
     sas = nil
+
     source.with_provider_connection do |azure|
       sas = Azure::Armrest::StorageAccountService.new(azure)
     end
+
     return if sas.nil?
 
     begin
-      key             = sas.list_account_keys(storage_account_name, storage_account_resource_group).fetch('key1')
-      storage_account = sas.get(storage_account_name, storage_account_resource_group)
-      blob            = storage_account.blobs("system", key).first
-      blob_properties = storage_account.blob_properties(blob.container, blob.name, key)
-      endpoint        = storage_account.properties.primary_endpoints.blob
-      source_uri      = File.join(endpoint, blob.container, blob.name)
-      target_uri      = File.join(endpoint, "manageiq", dest_name + "_" + SecureRandom.uuid + ".vhd")
+      image = sas.list_private_images.first 
+      return unless image
```

Comment:
Ah, I may need to put those methods back then and filter based on the storage account details.